File: Java/nbodysimulation/Integrators.java
package nbodysimulation;

public class Integrators{

    public static Body[] SymplecticEuler(Body[] bodies, double dt) {
        for (Body body : bodies) {
            body.calculateAcceleration(bodies);
            body.setVelocity(Vector.add(body.getVelocity(), Vector.multiply(body.getAcceleration(), dt)));
            body.setPosition(Vector.add(body.getPosition(), Vector.multiply(body.getVelocity(), dt)));
        }
        return bodies;
    }

    public static Body[] Euler(Body[] bodies, double dt) {
        for (Body body : bodies) {
            body.calculateAcceleration(bodies);
            body.setPosition(Vector.add(body.getPosition(), Vector.multiply(body.getVelocity(), dt)));
            body.setVelocity(Vector.add(body.getVelocity(), Vector.multiply(body.getAcceleration(), dt)));
        }
        return bodies;
    }

}

File: CPlusPlus/Integrators.cpp
#include <iostream>
#include <vector>
#include "Body.h"
#include "Integrators.h"

std::vector<Body*> Integrators::symplecticEuler(std::vector<Body*>& bodies, float dt) {
    for (Body* body : bodies) {
        body->calculateAcceleration(bodies);
    }
    for (Body* body : bodies) {
        body->setVelocity(body->getVelocity() + body->getAcceleration().scalarMultiply(dt));
        body->setPosition(body->getPosition() + body->getVelocity().scalarMultiply(dt));
    }
    return bodies;
}